### ISSUE TITLE
Accept array for spreadInsert

### ISSUE THREAD

**Issue by tjoskar** [andywer/squid#19]
Hi, I just found this library and it looks really good! 

I just miss one feature before I can start using it for real. I want to be able to do something like:

```ts
const users = await database.query(sql`
  INSERT INTO users ${spreadInserts([
    { name: "Jon", email: "jon.snow@winterfell.com" },
    { name: "Bran", email: "bran.stark@winterfell.com" }
  ])}
`)

// Text: INSERT INTO users (name, email) VALUES (%1, %2), (%3, %4)
// Values: [ "Jon", "jon.snow@winterfell.com", "Bran", "bran.stark@winterfell.com" ]
``` 

I can submit a PR if you want.

**Comment by andywer** [comment]
Hey @tjoskar!

Yeah, that makes a lot of sense 👍
I am just wondering about the API. Should we add `spreadInserts()` or should we overload `spreadInsert()`, so that you can also pass an array of objects to it? 🤔

**Comment by tjoskar** [comment]
> Thanks for your PR. Let's discuss this first :)

Absolutely, sorry for my eager ;) 

> I am just wondering about the API. Should we add spreadInserts() or should we overload spreadInsert(), so that you can also pass an array of objects to it?

I was thinking of having two different functions at first, something like this:
```ts
function spreadInserts(records: any[]) {
  // logic
}

function spreadInsert(record: any) {
  return spreadInserts([record])
}

spreadInsert({ ... })
spreadInserts([{ ... }, { ... }])
```

But I think it make more sense to accept arbitrary number of arguments:

```ts
function spreadInsert(...records: any[]) {
  // logic
}
spreadInsert({ ... })
spreadInsert([{ ... }, { ... }])
```

Do you agree?

> When you say you see the errors that you describe, do you mean postguard throws them or is it a runtime error?

I mean runtime error. In the same way you are getting the error `Error: Invalid identifier: some"column` https://github.com/andywer/squid/blob/2ac01c35aa5160165acb3717e2e4356253a41201/src/sql.ts#L20

Regarding my errors. The main problem is how we should handle cases like:

```
sql`INSERT INTO users ${spreadInsert({
    name: "Hugo",
    age: 20
  }, {
    title: "Mr",
    surname: "Jonson"
  })} RETURNING *`
```

What is expected behavior here? I see a couple different options:

1. Use the first record as the truth and only use the column `name` and `age`. Throw an error that `name` and `age` is missing on the second record.

2. Use the first record as the truth and only use the column `name` and `age`, use "default" for the second record: `INSERT INTO users (name, age) VALUES ($1, $2), (default, default)`, `values: ["Hugo", 20]`

3. Merge all columns name and use default for the missing ones: `INSERT INTO users (name, age, title, surname) VALUES ($1, $2, default, default), (default, default, $3, $4)`, `values: ["Hugo", 20, "Mr", "Jonson"]`

I'm voting for 1 (as it is right now). I think it is dangerous to assume that the user wants to use the `default` value. If one wants to use the default value, one should state that:
```ts
sql`INSERT INTO users ${spreadInsert({
    name: "Hugo",
    age: 20,
    title: sql.raw("dafault"),
    surname: sql.raw("dafault")
  }, {
    name: sql.raw("dafault"),
    age: sql.raw("dafault"),
    title: "Mr",
    surname: "Jonson"
  })} RETURNING *`
```

**Comment by andywer** [comment]
Yeah, that sounds good! Like it 👍

**Comment by andywer** [comment]
Closed by #20. Accepting multiple arguments, not an array, though.
